### PR TITLE
[RFC] test: Add arm64_high_addresses test

### PR DESCRIPTION
**Pull request by carlocaione**
First of all I'm not exactly sure if this is the best way to tackle this problem but at least it's something. @nashif (or any other CI expert) please let me know if you think there is a better way to do this.

So, the problem is that in the past couple of months we hit several time several issues related to having the SRAM to high addresses in memory or when `SRAM_BASE_ADDRESS != KERNEL_VM_BASE`: see #33607, #33594, #33204, #32053.

We do not have a practical way to actual test zephyr in emulation when the SRAM is mapped to high addresses but at least we can try to catch early problems at compile time that would have spared a bit of head-scratch in the past.

So this test is basically the build-only copy of the `hello_world_user` test but using high addresses for `SRAM_BASE_ADDRESS` and `KERNEL_VM_BASE` in several combinations.

### CODE REVIEW

**Comment by carlocaione** [comment]
Of course this is failing on CI because #33594 has not been merged yet (and this proves that this works ;)

**Comment by npitre** [comment]
I wonder if `arch_allow: arm` won't let ARM32 through where this test is 
definitely wrong.

And if we're going to have "low" addresses, I'd use something actually 
low i.g. 0x400000 instead of 0x40000000.

**Comment by carlocaione** [comment]
> I wonder if `arch_allow: arm` won't let ARM32 through where this test is definitely wrong.

The `filter: CONFIG_64BIT` should be enough to avoid building for ARM32

> And if we're going to have "low" addresses, I'd use something actually low i.g. 0x400000 instead of 0x40000000.

Yup, makes sense.

**Review by npitre — APPROVED** [review]

**Comment by cfriedt** [comment]
@carlocaione - can you rebase? I see nothing wrong with additional tests.

**Comment by carlocaione** [comment]
@cfriedt can we merge this?

File: tests/arch/arm64/arm64_high_addresses/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.13.1)
```
Comment (stephanosio) [inline tests/arch/arm64/arm64_high_addresses/CMakeLists.txt:3]:
```suggestion
cmake_minimum_required(VERSION 3.20.0)
```

Need 3.20.0 now (see #37575).

File: tests/arch/arm64/arm64_high_addresses/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+
+include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
```
Comment (stephanosio) [inline tests/arch/arm64/arm64_high_addresses/CMakeLists.txt:5]:
```suggestion
find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
```

We use CMake packages now.

File: tests/arch/arm64/arm64_high_addresses/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+project(hello_world_user)
```
Comment (stephanosio) [inline tests/arch/arm64/arm64_high_addresses/CMakeLists.txt:6]: Maybe rename it to match the actual test name?

```suggestion
project(arm64_high_addresses)
```

**Review by stephanosio — COMMENTED** [review]
Looks ok overall. Some comments about the CMakeLists.

File: tests/arch/arm64/arm64_high_addresses/testcase.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+common:
+  filter: CONFIG_64BIT
+  arch_allow: arm64
```
Comment (stephanosio) [inline tests/arch/arm64/arm64_high_addresses/testcase.yaml:3]: Do we need `filter: CONFIG_64BIT` if we specify `arch_allow: arm64`?

File: tests/arch/arm64/arm64_high_addresses/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# SPDX-License-Identifier: Apache-2.0
+
+cmake_minimum_required(VERSION 3.20.1)
```
Comment (stephanosio) [inline tests/arch/arm64/arm64_high_addresses/CMakeLists.txt:3]: We have 3.20.0 for the rest of the tests and samples.
```suggestion

cmake_minimum_required(VERSION 3.20.0)
```

File: tests/arch/arm64/arm64_high_addresses/testcase.yaml (hunk 4 above)
Comment (carlocaione) [reply tests/arch/arm64/arm64_high_addresses/testcase.yaml:3]: right. This PR is so old that at that time there was no arm64 architecture

**Review by dcpleung — APPROVED** [review]

**Pull request merged**